tests/pruners_tests/test_successive_halving.py

```diff
@@ -108,6 +108,34 @@ def test_successive_halving_pruner_with_nan():
     assert pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id))
 
 
+def test_successive_halving_pruner_with_auto_min_resource():
+    # type: () -> None
+
+    pruner = optuna.pruners.SuccessiveHalvingPruner(min_resource='auto')
+    study = optuna.study.create_study(sampler=optuna.samplers.RandomSampler(), pruner=pruner)
+
+    assert pruner._min_resource is None
+
+    def objective(trial):
+        # type: (optuna.trial.Trial) -> float
+
+        for i in range(3):
```

Comment:
How about parameterizing the number of steps to cover edge cases incl. 0 and 1?

Reply:
Sure thing.
I'll parametrize the number of reports and trials.